### PR TITLE
fisrst lab 

### PR DESCRIPTION
**Pull request by ix21027**

### CODE REVIEW

File: lab.js [hunk 1]
```diff
@@ -71,14 +85,17 @@ console.log(p.getReviewByID(1).comment);
 console.log(p.getImage("url/img1"));
 console.log(p.addSize("XXS"));
 console.log(p.getSizes());
+console.log(p.setActiveSize('XS'));
+console.log(p.getActiveSize());
 console.log(p.deleteSize("XS"));
 console.log(p.getSizes());
-console.log(p.getAverageRating());
+console.log(p.getAverageRating('price'));
```
Comment (lzviryka) [inline lab.js]: clean up your checking and minimize comments

File: lab.js [hunk 2]
```diff
@@ -71,14 +85,17 @@ console.log(p.getReviewByID(1).comment);
...
 
 
 let searchProducts = (products, query) => {
     products.filter(p => p.getName.includes(query)  || p.getDescription.includes(query))
 }
+
 let sortProducts = (products, sortRule = "price") => {
     if (sortRule === "price" || sortRule === "id")
```
Comment (lzviryka) [inline lab.js]: typeof id === 'string'

File: lab.js [hunk 3]
```diff
@@ -52,16 +63,19 @@ function Product(params = { name: "", description: "", price: 1, brand: "", quan
     this.deleteReview = (id) => this.reviews = this.reviews.filter(r => r.ID !== id)
 
     this.deleteSize = (that) => this.sizes = this.sizes.filter(size => size !== that)
-    this.getAverageRating = () => _.meanBy(this.reviews, r => r.rating.value)
+    this.getAverageRating = (key) => _.meanBy(this.reviews, r => r.rating[key])
```
Comment (lzviryka) [inline lab.js]: You must return object
object = {service: avarageService ,
                 price: avaragePrice,
                 value: avarageValue,
                  quality: avarageQuality,
                  }

File: lab.js [hunk 4]
```diff
@@ -1,9 +1,20 @@
 var _ = require('lodash'); 
 
-function Product(params = { name: "", description: "", price: 1, brand: "", quantity: 1, 
-                            date: new Date(), reviews: [], images: [], activeSize: "",
-                            sizes: ['XS', 'S', 'M', 'L', 'XL', 'XXL'], ID: (+new Date()+"" )}
-                ){
+function Review(data = {}){
+    const {ID, author, date, comment, rating} = data;
```
Comment (lzviryka) [inline lab.js]: for what this code?

File: lab.js [hunk 5]
```diff
@@ -1,9 +1,20 @@
 var _ = require('lodash'); 
 
-function Product(params = { name: "", description: "", price: 1, brand: "", quantity: 1, 
-                            date: new Date(), reviews: [], images: [], activeSize: "",
-                            sizes: ['XS', 'S', 'M', 'L', 'XL', 'XXL'], ID: (+new Date()+"" )}
-                ){
+function Review(data = {}){
```
Comment (lzviryka) [inline lab.js:5]: if you make empty object, then you need the default value

File: lab.js [hunk 6]
```diff
@@ -1,9 +1,20 @@
...
+    this.comment = comment, 
+    this.rating = {
+        service: rating.service, 
+        price: rating.price, 
+        value:  rating.value, 
+        quality: rating.quality
+    }
+}
+
+function Product(params = { }){
```
Comment (lzviryka) [inline lab.js:21]: if you make empty object, then you need the default value

File: lab.js [hunk 7]
```diff
@@ -1,87 +1,73 @@
...
+        service: service || 0, 
+        price: price || 0, 
+        value:  value || 0, 
+        quality: quality || 0
     }
-    defineGetProps();
+}
+
+function Product(params = { }){
+    let {sizes, name, description, price, brand, quantity, date, reviews, images, activeSize, ID} = params
```
Comment (lzviryka) [inline lab.js:22]: for what this code?

File: lab.js [hunk 8]
```diff
@@ -1,87 +1,73 @@
...
+    this.sizes       = sizes || [];
+    this.name        = name || '';
```
Comment (lzviryka) [inline lab.js:25]: this.sizes       = params.sizes || [];

File: lab.js [hunk 9]
```diff
@@ -1,87 +1,73 @@
...
+    this.description = description || '';
+    this.price       = +price || 1;
```
Comment (lzviryka) [inline lab.js]: default price should be 0

File: lab.js [hunk 10]
```diff
@@ -1,87 +1,73 @@
...
+    this.brand       = brand || '';
+    this.quantity    = +quantity || 1;
+    this.date        = date || new Date();
+    this.reviews     = reviews || [];
+    this.images      = images || [];
+    this.activeSize  = activeSize || '';
+    this.ID          = ID || (+new Date()+"" );
```
Comment (lzviryka) [inline lab.js:34]: if there is the same id?

File: lab.js (hunk 7 above)
Comment (ix21027) [reply lab.js:22]: https://github.com/airbnb/javascript#destructuring

**Pull request merged**